PyVista fails at creating a pv.Rectangle

### Describe the bug, what's wrong, and what you expected.

In some circumstances, pv.Rectangle constructor called with 3 points fails at creating the rectangle with the following error:

```python
r = pv.Rectangle(
    [
        [0.0436306, 0.035991, 0.0001723],
        [0.0446907, 0.035991, 0.0001723],
        [0.0436306, 0.036061, 0.0001723],
    ]
)
```

pv_test.py:6: 

points = array([[0.0436306, 0.035991 , 0.0001723],
       [0.0446907, 0.035991 , 0.0001723],
       [0.0436306, 0.036061 , 0.0001723]])

    def Rectangle(points=None):
        """Create a rectangle defined by 3 points.
    
        The 3 points must define an orthogonal set of vectors.
    
        Parameters
        ----------
        points : array_like[float], optional
            Points of the rectangle. Defaults to a unit square in xy-plane.
    
        Returns
        -------
        pyvista.PolyData
            Rectangle mesh.
    
        Examples
        --------
        >>> import pyvista as pv
        >>> pointa = [1.0, 0.0, 0.0]
        >>> pointb = [1.0, 1.0, 0.0]
        >>> pointc = [0.0, 1.0, 0.0]
        >>> rectangle = pv.Rectangle([pointa, pointb, pointc])
        >>> rectangle.plot(show_edges=True, line_width=5)
        """
        if points is None:
            points = [[1.0, 0.0, 0.0], [1.0, 1.0, 0.0], [0.0, 1.0, 0.0]]
        if len(points) != 3:
            raise TypeError('Points must be given as length 3 np.ndarray or list')
    
        points, _ = _coerce_pointslike_arg(points)
    
        point_0 = points[0]
        point_1 = points[1]
        point_2 = points[2]
    
        vec_01 = point_1 - point_0
        vec_02 = point_2 - point_0
        vec_12 = point_2 - point_1
    
        scalar_pdct_01_02 = np.dot(vec_01, vec_02)
        scalar_pdct_01_12 = np.dot(vec_01, vec_12)
        scalar_pdct_02_12 = np.dot(vec_02, vec_12)
    
        null_scalar_products = [
            val
            for val in [scalar_pdct_01_02, scalar_pdct_01_12, scalar_pdct_02_12]
            if np.isclose(val, 0)
        ]
        if len(null_scalar_products) == 0:
            raise ValueError("The three points should defined orthogonal vectors")
        if len(null_scalar_products) > 1:
           raise ValueError("Unable to build a rectangle with less than three different points")
E           ValueError: Unable to build a rectangle with less than three different points

This happens with at least PyVista 0.42.0 and 0.43.3.

### Steps to reproduce the bug.

```python
import pyvista as pv


def test_pv_rectagle() -> None:
    r = pv.Rectangle(
        [
            [0.0436306, 0.035991, 0.0001723],
            [0.0446907, 0.035991, 0.0001723],
            [0.0436306, 0.036061, 0.0001723],
        ]
    )
    print(r)

```

### System Information

```shell
--------------------------------------------------------------------------------
  Date: Wed Mar 06 10:31:50 2024 CET

                OS : Linux
            CPU(s) : 4
           Machine : x86_64
      Architecture : 64bit
       Environment : Python
        GPU Vendor : Intel
      GPU Renderer : Mesa Intel(R) HD Graphics 620 (KBL GT2)
       GPU Version : 4.6 (Core Profile) Mesa 21.2.6
  MathText Support : True

  Python 3.9.5 (default, Nov 23 2021, 15:27:38)  [GCC 9.3.0]

           pyvista : 0.43.3
               vtk : 9.3.0
             numpy : 1.26.4
        matplotlib : 3.8.3
            scooby : 0.9.2
             pooch : 1.8.1
            pillow : 10.2.0
--------------------------------------------------------------------------------
```


### Screenshots

_No response_

--- a/pyvista/core/utilities/geometric_objects.py
+++ b/pyvista/core/utilities/geometric_objects.py
@@ -1907,9 +1907,16 @@ def Rectangle(points=None):
     vec_02 = point_2 - point_0
     vec_12 = point_2 - point_1
 
-    scalar_pdct_01_02 = np.dot(vec_01, vec_02)
-    scalar_pdct_01_12 = np.dot(vec_01, vec_12)
-    scalar_pdct_02_12 = np.dot(vec_02, vec_12)
+    mag_01 = np.linalg.norm(vec_01)
+    mag_02 = np.linalg.norm(vec_02)
+    mag_12 = np.linalg.norm(vec_12)
+
+    if np.isclose(mag_01, 0) or np.isclose(mag_02, 0) or np.isclose(mag_12, 0):
+        raise ValueError("Unable to build a rectangle with less than three different points")
+
+    scalar_pdct_01_02 = np.dot(vec_01, vec_02) / min(mag_01, mag_02) ** 2
+    scalar_pdct_01_12 = np.dot(vec_01, vec_12) / min(mag_01, mag_12) ** 2
+    scalar_pdct_02_12 = np.dot(vec_02, vec_12) / min(mag_02, mag_12) ** 2
 
     null_scalar_products = [
         val
@@ -1918,8 +1925,6 @@ def Rectangle(points=None):
     ]
     if len(null_scalar_products) == 0:
         raise ValueError("The three points should defined orthogonal vectors")
-    if len(null_scalar_products) > 1:
-        raise ValueError("Unable to build a rectangle with less than three different points")
 
     points = np.array([point_0, point_1, point_2, point_0])
     if np.isclose(scalar_pdct_01_02, 0):

--- a/tests/core/test_geometric_objects.py
+++ b/tests/core/test_geometric_objects.py
@@ -597,11 +597,20 @@ def test_quadrilateral():
     assert np.allclose(mesh.points, points)
 
 
-def test_rectangle():
-    pointa = [3.0, 1.0, 1.0]
-    pointb = [3.0, 2.0, 1.0]
-    pointc = [1.0, 2.0, 1.0]
-    pointd = [1.0, 1.0, 1.0]
+@pytest.mark.parametrize(
+    "points",
+    [
+        ([3.0, 1.0, 1.0], [3.0, 2.0, 1.0], [1.0, 2.0, 1.0], [1.0, 1.0, 1.0]),
+        (
+            [0.043, 0.0359, 0.0001],
+            [0.044, 0.0359, 0.0001],
+            [0.043, 0.036, 0.0001],
+            [0.044, 0.036, 0.0001],
+        ),
+    ],
+)
+def test_rectangle(points):
+    pointa, pointb, pointc, pointd = points
 
     # Do a rotation to be in full 3D space with floating point coordinates
     trans = pv.core.utilities.transformations.axis_angle_rotation([1, 1, 1], 30)

